### ISSUE TITLE
Make the `Collect` trait part of the public api

### ISSUE THREAD

**Issue by konstin** [dtolnay/inventory#5]
It's currently not possible to write a function generic over `T`where `T` is an inventory.

To rewrite the example given in the docs page:

```rust
fn print_inventory<T: inventory::Collect>() {
    for flag in inventory::iter::<T> {
        println!("-{}, --{}", flag.short, flag.name);
    }
}
```

This does already work, but the `Collect` trait is currently `#[doc(hidden)]`. My specific use case is doing the following, combined with two proc macros and a helper traits:

```rust
/// Allows arbitrary pymethod blocks to submit their methods, which are eventually collected by pyclass
pub trait PyMethodsInventory: inventory::Collect {
    /// Create a new instance
    fn new(methods: &'static [PyMethodDefType]) -> Self;

    /// Returns the methods for a single impl block
    fn get_methods(&self) -> &'static [PyMethodDefType];
}
```

**Comment by dtolnay** [comment]
Fixed in 0.1.3.

**Comment by konstin** [comment]
Thank you!
